File: scratch/replace_years.js
const fs = require('fs');
const path = require('path');

const rootDir = path.join(__dirname, '..', 'app', '(main)');

let filesModified = 0;
let totalReplacements = 0;

function walkDir(currentDir) {
    const files = fs.readdirSync(currentDir);
    for (const file of files) {
        const filePath = path.join(currentDir, file);
        const stat = fs.statSync(filePath);
        if (stat.isDirectory()) {
            walkDir(filePath);
        } else if (stat.isFile() && (file.endsWith('.ts') || file.endsWith('.tsx') || file.endsWith('.json') || file.endsWith('.xml'))) {
            try {
                let content = fs.readFileSync(filePath, 'utf8');
                if (content.includes('2025')) {
                    const occurrences = content.split('2025').length - 1;
                    content = content.replace(/2025/g, '2026');
                    fs.writeFileSync(filePath, content, 'utf8');
                    console.log(`Updated ${filePath}: replaced ${occurrences} occurrence(s)`);
                    filesModified++;
                    totalReplacements += occurrences;
                }
            } catch (err) {
                console.error(`Error processing ${filePath}:`, err);
            }
        }
    }
}

walkDir(rootDir);
console.log(`\nDone! Modified ${filesModified} file(s) with ${totalReplacements} total replacement(s).`);

File: scratch/replace_years.py
import os

root_dir = r"d:\Taxi KSA\taxi-service-ksa\app\(main)"

files_modified = 0
total_replacements = 0

for dirpath, dirnames, filenames in os.walk(root_dir):
    for filename in filenames:
        if filename.endswith(('.ts', '.tsx', '.json', '.xml')):
            filepath = os.path.join(dirpath, filename)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                if "2025" in content:
                    count = content.count("2025")
                    new_content = content.replace("2025", "2026")
                    
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    
                    print(f"Updated {filepath}: replaced {count} occurrence(s)")
                    files_modified += 1
                    total_replacements += count
            except Exception as e:
                print(f"Error reading/writing {filepath}: {e}")

print(f"\nDone! Modified {files_modified} file(s) with {total_replacements} total replacement(s).")
